impl/src/settings.rs

```diff
@@ -367,8 +429,90 @@ pub fn format_registry_url(registry_url: &str, name: &str, version: &str) -> Str
     .replace("{version}", version)
 }
 
+fn validate_create_setting_additional_build_file_is_relative(
+  file_path: &Path,
+  crate_name: &str,
+  version: &VersionReq,
+) -> Result<(), RazeError> {
+  if file_path.is_absolute() {
+    return Err(RazeError::Config {
+      field_path_opt: Some(format!(
+        "raze.crates.{}.{}.additional_build_file",
+        crate_name,
+        version.to_string()
+      )),
+      message: "additional_build_file flags should be reletive paths from the manifest containing \
+                the definition"
+        .to_owned(),
+    });
+  }
+
+  Ok(())
+}
+
+fn validate_create_setting_additional_build_file(
+  additional_build_file: &mut &mut AdditionalBuildFile,
+  toml_path: &Path,
+  crate_name: &str,
+  version: &VersionReq,
+) -> Result<(), RazeError> {
+  match additional_build_file {
+    AdditionalBuildFile::Raw(name) => {
+      **additional_build_file = {
+        let file_path = PathBuf::from(&name);
+        validate_create_setting_additional_build_file_is_relative(&file_path, crate_name, version)?;
+
+        // Canonicalize the path of the additional build file
+        match toml_path.join(&file_path).canonicalize() {
+          Ok(path) => AdditionalBuildFile::Validated {
+            name: name.clone(),
+            path,
+          },
+          Err(err) => {
+            return Err(RazeError::Config {
+              field_path_opt: Some(format!(
+                "raze.crates.{}.{}.additional_build_file",
+                crate_name,
+                version.to_string()
+              )),
+              message: format!(
+                "Failed to canonicalize string with error: {}",
+                err.to_string()
+              ),
+            });
+          },
+        }
+      }
+    },
+    _ => { /* Do nothing for any other context */ },
+  }
+
+  Ok(())
+}
+
+/** Ensures crate settings which specify additional build files have canonicalized paths */
+fn validate_crate_settings(settings: &mut RazeSettings, toml_path: &Path) -> Result<(), RazeError> {
+  for (crate_name, crate_settings) in settings.crates.iter_mut() {
+    for (version, crate_settings) in crate_settings.iter_mut() {
+      if crate_settings.additional_build_file.is_none() {
+        continue;
+      }
+
+      let additional_build_file: &mut &mut AdditionalBuildFile =
+        &mut crate_settings.additional_build_file.as_mut().unwrap();
+      validate_create_setting_additional_build_file(
+        additional_build_file,
+        toml_path,
+        crate_name,
+        version,
+      )?;
```

Comment:
There isn't one currently (that I know of, unless `anyhow` has some functionality for that).  I feel generally that the `error` module needs to be revisited. Each case is kinda clunky (given the common use of `anyhow` and I think we should either move more into `anyhow` or refine the Errors. I'd rather not introduce a new crate just for this since I feel like with some refactoring it won't be necessary. If you feel strongly about it I can try to crack it out quick but I have a queue of changes I'd rather get in first.

Reply:
`check_all_vendored` does something manual that you could consider here. It aggregates the errors as crate names, then converts them to a single error.

I don't really feel strongly about this though.

Reply:
I've updated this but I'm not happy about the state of the state of it. I totally understand the value in collecting all errors vs surfacing one at a time but I think the correct solution here is to update `RazeError` to handle chained errors. Having to collect errors then write them to a string seems to invalidate the existence of `RazeError` in that e might as well be using `anyhow` everywhere if the only interesting content is the message. I'd much prefer for the errors themselves to carry more meaning and have relevant and convenient data associated that might even be able to convey the message on it's own.

Anyway, updated